addOns/retest/src/main/java/org/zaproxy/addon/retest/ExtensionRetest.java

```diff
@@ -131,4 +136,21 @@ public static boolean testsForAlert(AutomationAlertTest test, AlertData data) {
                 && data.getRisk().equals(testData.getRisk())
                 && data.getOtherInfo().equals(testData.getOtherInfo());
     }
+
+    private class SessionChangedListenerImpl implements SessionChangedListener {
+
+        @Override
+        public void sessionScopeChanged(Session session) {}
+
+        @Override
+        public void sessionModeChanged(Mode mode) {}
+
+        @Override
+        public void sessionChanged(Session session) {}
+
+        @Override
+        public void sessionAboutToChange(Session session) {
+            retestDialog = null;
```

Comment:
Better `dispose`, if the dialogue is open it would be kept around with outdated info.